graphlearn/src/core/operator/sampler/alias_method.h

```diff
@@ -52,6 +52,24 @@ class AliasMethodFactory {
     static AliasMethodFactory factory;
     return &factory;
   }
+
+  template<class T>
+  AliasMethod* LookupOrCreate(const std::string& type,
+      const io::Array<float> weights) {
+    ScopedLocker<std::mutex> _(&mtx_);
+    auto it = map_.find(type);
+    if (it == map_.end()) {
+      std::vector<float> tmp_w(weights.Size());
+      for (size_t idx = 0; idx < weights.Size(); ++idx) {
+        tmp_w[idx] = weights[idx];
+      }
+      auto am = new AliasMethod(&tmp_w);
+      map_[type] = am;
+      return am;
+    } else {
+      return it->second;
+    }
+  }
   
   template<class T>
   AliasMethod* LookupOrCreate(const std::string& type, 
```

Comment:
Duplicate with line 57 `LookupOrCreate` ?

Reply:
It is an overloading, not a duplication. See my explaination for `condition_table.h`.